[Gavin Goh Jie Wei] iP

gavin331 opened the pull request:
# BartenderBob
> "The purpose of software engineering is to control complexity, and not to create it." - Jon L. Bentley ([source](https://quotefancy.com/quote/2206958/Jon-L-Bentley-The-purpose-of-software-engineering-is-to-control-complexity-not-to-create))

BartenderBob chat bot frees your mind of having to remember things you need to do. It's,
+ text-based
+ easy to learn
+ ~~FAST~~ _SUPER FAST_ to use!

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2324s2.github.io/website/schedule/week4/project.html).
1. double-click it.
1. add your tasks.
1. let it manage your tasks for you 😀 

AND it is **FREE**!

Features:

- [X] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

gavin331 committed:
Implemented Level-0

gavin331 committed:
Added BartenderBob class

gavin331 committed:
Added storage and its functions

gavin331 committed:
Added InputHandler and Task classes

gavin331 committed:
Added Deadline and Event classes

gavin331 committed:
Filled in test cases in expected.txt and input.txt

gavin331 committed:
Created BartenderBobException to handle exceptions

gavin331 committed:
Implemented delete, updated tests and exceptions

gavin331 committed:
Implement save feature

gavin331 committed:
Fix bugs from save feature

gavin331 committed:
Merge branch 'branch-Level-7'

gavin331 committed:
Implement dates and times

gavin331 committed:
Merge branch 'branch-Level-8'

gavin331 committed:
Improve OOP

gavin331 committed:
Remove unused code

gavin331 committed:
Add BartenderBob package

gavin331 committed:
Merge branch 'add-gradle-support' of https://github.com/gavin331/ip

gavin331 committed:
Add gradle

gavin331 committed:
Handle invalid date format

gavin331 committed:
Add Junit test cases

gavin331 committed:
Create jar file

gavin331 committed:
Add Java Docs

gavin331 committed:
Merge branch 'branch-A-JavaDoc'

gavin331 committed:
Improve coding standards

gavin331 committed:
Merge branch 'A-CodingStandard'

gavin331 committed:
Implement find feature

gavin331 committed:
Merge branch 'Level-9'

gavin331 committed:
Implement checkstyle

gavin331 committed:
Merge branch 'branch-A-CheckStyle'

gavin331 committed:
Implement Javafx GUI

gavin331 committed:
Complete Javafx GUI

gavin331 committed:
Merge branch 'branch-Level-10'

gavin331 committed:
Update javafx UI

jeong-jaeho commented on src/main/java/bartenderbob/Deadline.java:
```diff
@@ -0,0 +1,72 @@
+package bartenderbob;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
```
I like how you avoided using the wildcard (*) and instead imported specific classes you need!

jeong-jaeho commented on src/main/java/bartenderbob/Deadline.java:
```diff
@@ -0,0 +1,72 @@
+     *
+     * @return Complete deadline task information as a String.
+     */
+    @Override
```
I like the way you have overriden the show() method to make it specific to the deadline class.

jeong-jaeho commented on src/main/java/bartenderbob/DialogBox.java:
```diff
@@ -0,0 +1,61 @@
+     * Flips the dialog box such that the ImageView is on the left and text on the right.
+     */
+    private void flip() {
+        ObservableList<Node> tmp = FXCollections.observableArrayList(this.getChildren());
```
I like how you practiced good practice of correct indentation here. It's very easy to read.

jeong-jaeho commented on src/main/java/bartenderbob/BartenderBob.java:
```diff
@@ -0,0 +1,38 @@
+ */
+public class BartenderBob {
+    /** Name of the chatbot */
+    private static final String NAME = "BartenderBob";
```
I like how you use all uppercase for constant names! You also abide well with camelCase naming for variables.

jeong-jaeho reviewed (commented):
LGTM. Keep on the good work!

gavin331 committed:
Implement assertion

gavin331 committed:
Overall code quality is not up to standard. Methods are too long with
deep nesting.

Refactoring the code to reduce both nesting and the number
of LOC helps improve abstraction and readability.

Let's refactor the InputHandler and Storage classes to
enhance our code quality.

Refactoring into separate and smaller methods follows the
Single Level Of Abstraction (SLAP) principle, as well as
reduce the LOC.

gavin331 committed:
Merge pull request #1 from gavin331/branch-A-Assertions

Implement assertion

gavin331 committed:
Merge branch 'master' into branch-A-CodeQuality

gavin331 committed:
Merge pull request #2 from gavin331/branch-A-CodeQuality

Improve code quality by refactoring

gavin331 committed:
Implement date clash feature

The current version of the chatbot does not check for date clashes for
the event task.

This is bad for the user because the user does not know that an
existing event clashes with the event the user wants to add.

Let's update the necessary classes in order to implement the check for
date clashes for event tasks.

peterXGD commented on src/main/java/bartenderbob/BartenderBob.java:
```diff
@@ -0,0 +1,40 @@
+     *
+     * @param filePath The file path of the stored tasks.
+     */
+    public BartenderBob(String filePath) {
```
I like that you have a good use of slap to keep the code easy to read.

peterXGD commented on src/main/java/bartenderbob/Deadline.java:
```diff
@@ -0,0 +1,73 @@
+     * @return Complete deadline task information as a String.
+     */
+    @Override
+    public String show() {
```
I like the way you use the variable status to check the state of the task your app tracks.

peterXGD commented on src/main/java/bartenderbob/Event.java:
```diff
@@ -0,0 +1,91 @@
+            return true;
+        } catch (Exception e) {
+            return false;
+        }
```
I like the way you handle errors by using assert in the private methods.

peterXGD commented on src/main/java/bartenderbob/InputHandler.java:
```diff
@@ -0,0 +1,168 @@
+        case "event":
+            return processEventCommand(userInput, taskList);
+        default:
+            return BartenderBobException.invalidInput(userInput);
```
I like that you keep all the cases togethers 👍 and compact, increasing readability

peterXGD commented on src/main/java/bartenderbob/Storage.java:
```diff
@@ -0,0 +1,180 @@
+        DateTimeFormatter outputFormatter = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+
+        // Format the LocalDate to the desired output pattern
+        return localDate.format(outputFormatter);
```
I like that you make multiple comments for confusing methods, making the code easy to read.

peterXGD commented on src/main/java/bartenderbob/Ui.java:
```diff
@@ -0,0 +1,75 @@
+package bartenderbob;
```
I like that you keep all the error messages in the Ui class, thus making a good use of abstraction and encapsulation.

peterXGD reviewed (commented):
LGTM💯 , keep it up.

gavin331 committed:
Add more javadocs

gavin331 committed:
Improve UI

gavin331 committed:
Add check for logical date inputs in Event

gavin331 committed:
Add BartenderBob title to GUI

gavin331 committed:
Implement junit test for Event

gavin331 committed:
Update README

gavin331 committed:
Remove :grinning: from README

gavin331 committed:
Update version to 0.3

gavin331 committed:
Add main method in BartenderBob

gavin331 committed:
Modify main class in build gradle

gavin331 committed:
Fix bug for Event class

gavin331 committed:
Improve javadocs
